lib/TPP/ConvertTppToXsmm.cpp

```diff
@@ -342,65 +342,85 @@ struct ConvertTppIdentityOp : public OpRewritePattern<tpp::IdentityOp> {
     computeBcastShapeInput(shapeOutput, shapeInput, bShapeInput);
     assert(shapeOutput.size() == bShapeInput.size());
     shapeInput = bShapeInput;
-
-    if (shapeInput[1] == 1 && shapeOutput[1] > 1) {
-      xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_ROW;
-      int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>(), 1);
-      return {ldi, bCast};
+    assert(shapeInput.size() == 1 || shapeInput.size() == 2);
+
+    // Handle 2d-case.
+    if (shapeInput.size() == 2) {
+      if (shapeInput[1] == 1 && shapeOutput[1] > 1) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_ROW;
+        int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>(), 1);
+        return {ldi, bCast};
+      }
+
+      if (shapeInput[0] == 1 && shapeOutput[0] > 1) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_COL;
+        int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>());
+        return {ldi, bCast};
+      }
+
+      if (shapeInput[0] == shapeOutput[0] && shapeInput[1] == shapeOutput[1]) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::NONE;
+        int64_t ldi = *getLeadingDim(inputType.cast<MemRefType>());
+        return {ldi, bCast};
+      }
     }
-
-    if (shapeInput[0] == 1 && shapeOutput[0] > 1) {
-      xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_COL;
-      int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>());
-      return {ldi, bCast};
-    }
-
-    if (shapeInput[0] == shapeOutput[0] && shapeInput[1] == shapeOutput[1]) {
-      xsmm::UnaryFlags bCast = xsmm::UnaryFlags::NONE;
-      int64_t ldi = *getLeadingDim(inputType.cast<MemRefType>());
-      return {ldi, bCast};
+    // Handle 1d-case.
+    if (shapeInput.size() == 1) {
+      if (shapeInput[0] == 1 && shapeOutput[0] > 1) {
```

Comment:
Isn't this case covered by the `BCAST_SCALAR` at the function's entry? (check `all_ones`)